### ISSUE TITLE
Generate and consume the package spec (.packages)

### ISSUE THREAD

**Issue by sethladd** [dart-lang/sdk#23372]
This is the main tracking bug to implement [the package spec proposal](https://github.com/dart-lang/dart_enhancement_proposals/issues/5). This proposal has been accepted (but has not yet gone through TC52), and work has begun on implementing it.

This is the main tracking bug. Each platform's specific bug is below:

- [x] VM: #23369 
- [x] VM: `--packages` doesn't deadlock: #23991
- [x] dart2js: #23371, first released in [1.12.0-dev.0.0](/dart-lang/sdk/releases/tag/1.12.0-dev.0.0)
- [x] Analyzer: #23373
- [x] Analysis Server: #23925 
- [x] Pub: global activate: dart-lang/pub#22
- [x] Pub: merge new version into sdk: #24094
- [x] Pub: dart-lang/pub#16, first released in [1.12.0-dev.0.0](/dart-lang/sdk/releases/tag/1.12.0-dev.0.0)
- [x] Fix self-references in Pub: dart-lang/pub#1294, will be released in 1.12.0-dev.3.0
- [x] WebStorm: https://youtrack.jetbrains.com/issue/WEB-16466
- [x] Fetching a `.packages` file after an `http:` entry point: #23952
- [x] Test package_config on the packagebots #24092 

**Note that additional work will be needed in 1.13**, see #24112

**Comment by sethladd** [comment]
dart2js has filed their merge request for this new functionality.

**Comment by nex3** [comment]
Since I don't see it linked anywhere here, the issue tracking the VM's implementation is #23369.

**Comment by sethladd** [comment]
Blocked by https://github.com/dart-lang/pub/issues/1294

**Comment by nex3** [comment]
@sethladd When adding new sub-issues to meta-issues, it would be nice to include them in the master list as well (I've done so for this one).

**Comment by sethladd** [comment]
Sure. I didn't know if you wanted blockers that came up after we filed the meta issue, since they weren't original implementation issues.

**Comment by nex3** [comment]
I think the main function of the meta-issue is to give people a single place to see "what is the current status of X", so having everything in the main list is a good idea. I'll mention it in the wiki entry.

**Comment by sethladd** [comment]
sgtm!

**Comment by mit-mit** [comment]
Meta bug, assigning to self to drive to completion for 1.12.

**Comment by mit-mit** [comment]
Marking #23373 as complete in checlist

**Comment by mit-mit** [comment]
Moved #24093 to 1.13

**Comment by mit-mit** [comment]
Created a new issue #24112 to tracks the second phase for this feature targeted for 1.13. Deleting 1.13 work items from present issue that is bound to 1.12.

**Comment by mit-mit** [comment]
Marking testing task complete as #24092 was closed.

**Comment by mit-mit** [comment]
Marking this feature complete for 1.12. We will expand on this in 1.13, see #24112

**Comment by sethladd** [comment]
Note, there are many open issues around the implementation of this feature. A sample:

https://github.com/dart-lang/sdk/issues/24126
https://github.com/dart-lang/sdk/issues/24127
https://github.com/dart-lang/sdk/issues/24122
https://github.com/dart-lang/sdk/issues/24116
